src/arbitrary.rs

```diff
@@ -248,6 +249,48 @@ impl_arb_for_tuples! {
     (H, 7),
 }
 
+impl<T: Arbitrary + Sized, const N: usize> Arbitrary for [T; N] {
+    #[allow(unused_variables)] // for [T; 0]
+    fn arbitrary(g: &mut Gen) -> [T; N] {
+        // The following code was adopted from the corresponding example provided for `MaybeUninit`:
+        // https://doc.rust-lang.org/1.50.0/core/mem/union.MaybeUninit.html#initializing-an-array-element-by-element
+
+        // SAFETY: Create an uninitialized array of `MaybeUninit`. The `assume_init` is
+        // safe because the type we are claiming to have initialized here is a
+        // bunch of `MaybeUninit`s, which do not require initialization.
+        let mut maybe_uninit: [MaybeUninit<T>; N] =
+            unsafe { MaybeUninit::uninit().assume_init() };
+
+        // SAFETY: Dropping a `MaybeUninit` does nothing. Thus using raw pointer
+        // assignment instead of `ptr::write` does not cause the old
+        // uninitialized value to be dropped. Also if there is a panic during
+        // this loop, we have a memory leak, but there is no memory safety
+        // issue.
```

Comment:
This wording is a little confusing to me. It says "Thus using raw pointer assignment _instead_ of `ptr::write`" and that makes me think that the code below is going to be using raw pointer assignment. But it in fact uses `ptr::write`.

It looks like the wording was copied from the example in std's docs, but the actual code here is using `ptr::write` instead of just `*elem = MaybeUninit::new(T::arbitrary(g))`. It seems like the latter would be preferrable?

Reply:
Fixed.